plugins/outputs/datadog/datadog.go

```diff
@@ -76,6 +76,10 @@ func (d *Datadog) Write(metrics []telegraf.Metric) error {
 
 	for _, m := range metrics {
 		if dogMs, err := buildMetrics(m); err == nil {
+			tagMap := m.Tags()
+			metricTags := buildTags(tagMap)
```

Comment:
This is much better but if you want to take it even further you could use TagList() to build the tags list.  TagList is always sorted so you can avoid the sort as well.